### PR TITLE
Feat/longest shortest trade duration

### PR DESCRIPTION
**Pull request by sheridavandenbent**
# Results of merging this

Creates shortest, average, longest trade length statistic, plus tests, for both portfolio and per pair.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "max-open-trades": 3,
   "starting-capital": 1000,
   "backtesting-from": "2021-03-11",
-  "backtesting-to": "2021-03-15",
+  "backtesting-to": "2021-05-15",
```
Comment (sheridavandenbent) [inline config.json:7]:
```suggestion
  "backtesting-to": "2021-03-15",
```

File: modules/output/results.py [hunk 2]
```diff
@@ -106,6 +111,11 @@ def show(self, currency_symbol: str):
         console.print(table_grid)
 
     def create_trade_info_table(self, justification) -> Table:
+        avg_trade_duration = pandas.Series(self.avg_trade_duration).dt.round('1s')[0].to_pytimedelta() \
+            if self.avg_trade_duration != timedelta(0) else '-'
```
Comment (geordle) [inline modules/output/results.py:115]: something like this:
```suggestion
        avg_trade_duration = timedelta(seconds=round(self.avg_trade_duration.total_seconds())) \
            if self.avg_trade_duration != timedelta(0) else '-'
```

Comment (geordle) [reply modules/output/results.py:115]: Or this? 
```suggestion
        avg_trade_duration = Timedelta(self.avg_trade_duration).round("1s").to_pytimedelta() \
            if self.avg_trade_duration != timedelta(0) else '-'
```

File: test/stats/test_main_results.py [hunk 3]
```diff
@@ -208,7 +209,7 @@ def test_dividing_assets():
 
 
 def test_n_trades():
-    """Given 'trades where made', 
+    """Given 'trades where made',
```
Comment (geordle) [inline test/stats/test_main_results.py:212]:
```suggestion
    """Given 'trades were made',
```
?

File: modules/output/results.py [hunk 4]
```diff
@@ -210,6 +225,11 @@ def show(instances: typing.List['CoinInsights'], currency_symbol: str):
         coin_signal_table = CoinInsights.create_coin_signals_table(justification)
 
         for c in instances:
+            avg_trade_duration = pandas.Series(c.avg_trade_duration).dt.round('1s')[0].to_pytimedelta() \
```
Comment (geordle) [inline modules/output/results.py:228]: These lines look similar to those above. Maybe move this to a seperate function?

**Review by geordle — CHANGES_REQUESTED** [review]
Nice!, just some aesthetics stuff.

File: test/stats/test_main_results.py (hunk 3 above)
Comment (sheridavandenbent) [reply test/stats/test_main_results.py:212]: Ohhh I can't believe I made a typo like that. Thanks!

File: modules/output/results.py (hunk 2 above)
Comment (sheridavandenbent) [reply modules/output/results.py]: Perfect! Went for the first one because it's better readable. Thank you!

File: modules/output/results.py (hunk 4 above)
Comment (sheridavandenbent) [reply modules/output/results.py]: Done!

**Review by geordle — APPROVED** [review]

**Pull request merged**